contribs/src/main/java/com/netflix/conductor/contribs/queue/amqp/AMQPObservableQueue.java

```
@@ -0,0 +1,568 @@
+				.contentType(settings.getContentType()).contentEncoding(settings.getContentEncoding())
+				.deliveryMode(settings.getDeliveryMode()).build();
+	}
+
+	private void publishMessage(Message message, String exchange, String routingKey) {
+		try {
+			final String payload = message.getPayload();
+			getOrCreateChannel().basicPublish(exchange, routingKey, buildBasicProperties(message, settings),
+					payload.getBytes(settings.getContentEncoding()));
+			logger.info(String.format("Published message to %s: %s", exchange, payload));
+		} catch (Exception ex) {
+			logger.error("Failed to publish message {} to {}", message.getPayload(), exchange, ex);
+			throw new RuntimeException(ex);
+		}
+	}
+
+	@Override
+	public void publish(List<Message> messages) {
+		try {
+			final String exchange, routingKey;
+			if (useExchange) {
+				// Use exchange + routing key for publishing
+				getOrCreateExchange(settings.getQueueOrExchangeName(), settings.getExchangeType(), settings.isDurable(),
+						settings.autoDelete(), settings.getArguments());
+				exchange = settings.getQueueOrExchangeName();
+				routingKey = settings.getRoutingKey();
+			} else {
+				// Use queue for publishing
+				final AMQP.Queue.DeclareOk declareOk = getOrCreateQueue(settings.getQueueOrExchangeName(),
+						settings.isDurable(), settings.isExclusive(), settings.autoDelete(), settings.getArguments());
+				exchange = StringUtils.EMPTY; // Empty exchange name for queue
+				routingKey = declareOk.getQueue(); // Routing name is the name of queue
+			}
+			messages.forEach(message -> publishMessage(message, exchange, routingKey));
+		} catch (final RuntimeException ex) {
+			throw ex;
+		} catch (final Exception ex) {
+			logger.error("Failed to publish messages: {}", ex.getMessage(), ex);
+			throw new RuntimeException(ex);
+		}
+
+	}
+
+	@Override
+	public void setUnackTimeout(Message message, long unackTimeout) {
+		// TODO Auto-generated method stub
```

Even though this method is not used at the moment, please provide an implementation or throw an `UnsupportedOperationException` to avoid gaps, if we have to use this method from ObservableQueue interface elsewhere in the future.